src/components/calcite-input/calcite-input.tsx

```diff
@@ -131,6 +131,13 @@ export class CalciteInput {
   /** input value */
   @Prop({ mutable: true, reflect: true }) value?: string = "";
 
+  @Watch("value") valueWatcher(): void {
+    this.calciteInputInput.emit({
+      element: this.childEl,
+      value: this.value
+    });
+  }
```

Comment:
Have you checked to make sure this doesn't cause an infinite loop when the `value` prop is being controlled?  It looks like this is the only event that emits the new value, so if the developer is setting the `value` attribute directly as a result of this event being emitted, then the watcher will fire once when `this.value` first changes inside `inputInputHandler` and then when the event is emitted, this might fire again and again each time `value` is set from outside.

Reply:
I think it should be ok because the watcher is only called if value is different. right?

Reply:
Theoretically yes, but would be good to check just to rule out possible race conditions.  The reason I brought this up is that I had to remove an event emitter from a watch method in radio-button because it was causing just this kind of issue.

Reply:
so would a check look like:

```
@Watch("value") valueWatcher(newValue: string, oldValue: string): void {
  if (newValue !== oldValue) {
    this.calciteInputInput.emit({
      element: this.childEl,
      value: this.value
    });
  }
}

```

Reply:
Oh, by check I mean try and use calcite-input in a JSX app.  This watcher wouldn't fire unless the new value differs from the old, so this check wouldn't do any good.

Reply:
Ah, ha. Yeah I thought that was the case with `@Watch`.

Reply:
Yeah, my hunch/concern is that depending on how fast the user is typing, there could be a sync/timing issue with the value being set internally first and then broadcasted after, which then sets it again from the outside, potentially causing a mismatch.

If there does end up being an issue with this in production apps, we might be able to setup some kind of reusable pattern that we could implement with other similar components where we detect if the `value` attribute is being controlled, which would refrain from storing the value internally and just rely on the event emitter to broadcast the new value, which the app would then set each time it changed from the top down, just as react does with its one-way data flow.

If the value attribute is not controlled, then the current behavior would take place where `this.value` is updated on each change and the event is still emitted.  However, I found for many years that the controlled approach is far superior because it eliminates duplicate local state that can get out of sync with external app state, but it would still make sense to support such a use case since that is how the native input element behaves.